### PR TITLE
Adding notifications for new messages

### LINKED ISSUE #207: Adding Notifications for New Chat Messages When Chat is Closed
Currently, when the chat is closed, there is no way for users to know if there are new messages without opening the chat. To improve the user experience, it would be helpful to add notifications or indicators that there are new messages in the chat when it is closed.
Expected Result:
When there are new messages in the chat and the chat window is closed, a notification or indicator should be displayed to let users know that there are new messages.

### PR DESCRIPTION
**Pull request by JonaQuaLabs**
fixes #207
Added a function in the chat component to open and close it, which also resets the unread message count to 0. Imported Chat and Close icons, as well as the Badge component to display the number of unread messages.

Added some styles to componentStyles.jsx to handle the chat's visibility state.

Test Notificacion

### CODE REVIEW

File: frontend/src/themes/componentsStyles.jsx
Comment (paoopintos) [inline frontend/src/themes/componentsStyles.jsx:1]: I believe it's better to put the styles of the Chat elements within the `Chat.jsx` component

Comment (JonaQuaLabs) [reply frontend/src/themes/componentsStyles.jsx:1]: Thanks for the suggestion, I already moved all the chat styles to Chat.jsx

**Review by paoopintos — APPROVED** [review]

File: frontend/src/components/Chat.jsx [hunk 1]
```diff
@@ -40,37 +56,133 @@ function Chat(props) {
...
+  transition: opacity 0.3s ease;
+`;
+const ChatContentWrapper = styled.div`
+  flex-grow: 1;
+  overflow-y: auto;
+  transition: opacity 0.3s ease;
+
+  ${(props) =>
+    props.hidden &&
+    css`
```
Comment (juanpqualabs) [inline frontend/src/components/Chat.jsx]: No need to add the 'css' function here

File: frontend/src/themes/componentsStyles.jsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import styled from "styled-components";
+import styled, { css } from "styled-components";
```
Comment (juanpqualabs) [inline frontend/src/themes/componentsStyles.jsx]: css is imported but not used

File: frontend/src/views/Room.jsx [hunk 3]
```diff
@@ -52,6 +52,7 @@ function Room() {
   const [open, setOpen] = useState(false);
   // create reference to access room state var in useEffect cleanup func
   const roomRef = useRef();
+  // eslint-disable-next-line no-undef
```
Comment (juanpqualabs) [inline frontend/src/views/Room.jsx]: is it strictly necessary to add this line?

**Comment by juanpqualabs** [comment]
I saw that when you open the chat the input box changes its size, could you fix that?

**Comment by JonaQuaLabs** [comment]
> I saw that when you open the chat the input box changes its size, could you fix that?

It's fixed, thank you for pointing out that detail.

**Review by juanpqualabs — APPROVED** [review]

**Review by fpyoung — APPROVED** [review]

**Pull request merged**